client/client.go

```diff
@@ -509,6 +509,58 @@ func (r *NotaryRepository) GetChangelist() (changelist.Changelist, error) {
 	return cl, nil
 }
 
+// GetDelegationRoles returns the keys and roles of the repository's delegations
+func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
+	// Update state of the repo to latest
+	_, err := r.Update()
+	if err != nil {
+		return nil, err
+	}
+
+	// All top level delegations (ex: targets/level1) are stored exclusively in targets.json
+	targetsJSON, err := r.fileStore.GetMeta("targets", 0)
+	if err != nil {
+		return nil, err
+	}
+	targets := &data.SignedTargets{}
+	err = json.Unmarshal(targetsJSON, targets)
+	if err != nil {
+		return nil, err
+	}
+	allDelegations := targets.Signed.Delegations.Roles
+
+	// make a copy for traversing nested delegations
+	delegationsList := make([]*data.Role, len(allDelegations))
+	copy(delegationsList, allDelegations)
+
+	// Now traverse to lower level delegations (ex: targets/level1/level2)
+	for len(delegationsList) > 0 {
+		// Pop off first delegation to traverse
+		delegation := delegationsList[0]
+		delegationsList = delegationsList[1:]
+
+		// Get metadata
+		delegationMetaJSON, err := r.fileStore.GetMeta(delegation.Name, 0)
+		// If we get an error, don't try to traverse further into this subtree because it doesn't exist or is malformed
+		if err != nil {
+			continue
+		}
+		delegationMeta := &data.SignedTargets{}
+		err = json.Unmarshal(delegationMetaJSON, delegationMeta)
+		if err != nil {
+			return nil, err
+		}
+		for _, d := range delegationMeta.Signed.Delegations.Roles {
```

Comment:
Yeah, it's nice :-) 

I actually saw this and knew I'd done something similar elsewhere so I had to go double check my code. I was doing map->slice conversion so unfortunately couldn't make use of this.